Chicago Weather API Class Structure

daisy-jg opened the pull request.

daisy-jg committed:
Created Weather API class structure

leonstaubach commented on weatherAPI.py:
```diff
@@ -0,0 +1,32 @@
+
+class WeatherAPIService(AbstractDataService):
+    def getWeatherForecast(self, latitude, longitude):
+        url = f"https://api.open-meteo.com/v1/forecast?latitude=41.85&longitude=-87.65&hourly=temperature_2m"
+        response = requests.get(url)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise Exception("Failed to fetch weather forecast")
```
few things here
- the url contains hard-coded request parameters instead of utilizing the passed in `latitude` and `longitude`
- the response is not parsed down enough (you simply return `response.json()`, instead find the specific fields in the response you need, and only return the values you find there (maybe with some validation)

leonstaubach commented on weatherAPI.py:
```diff
@@ -0,0 +1,32 @@
+
+class MockedWeatherService(AbstractDataService):
+    def getWeatherForecast(self, latitude, longitude):
+        return {
+            "latitude": latitude,
+            "longitude": longitude,
+            "hourly": {
+            }
```
you can add some value generation here (either with a constructor generator method or for now just simple hardcoded values)

leonstaubach reviewed (commented):
the class structure looks great, just as expected! Specifically for the `WeatherAPIService`, there are some updates necessary. You should also think about writing unit tests to verify that your methods return consistent data.

daisy-jg committed:
Updated WeatherAPIService class and MockedWeatherService class

• Fixed URL for the API request using the latitude and longitude parameters instead of hard-coded request parameters
• Updated to parse the API response by extracting the time and temperature from the response in the WeatherAPIService class
• Added sample data for hourly forecast in the MockedWeatherService class
